File: ContactListWebApp/Scripts/ContactListApp/Services/contactService.js
contactListApp.service("contactService",
    [
        "$http",
        "contactUrls",
        function ($http, contactUrls) {
            this.addContact = function(contact) {
                return $http(
                {
                    method: "POST", 
                    url: contactUrls.addContact, 
                    headers: { 'Content-Type': "application/json" },
                    data: contact
                });
            };

            this.updateContact = function (contact) {
                return $http(
                {
                    method: "POST",
                    url: contactUrls.updateContact,
                    headers: { 'Content-Type': "application/json" },
                    data: contact
                });
            };

            this.deleteContact = function (id) {
                return $http(
                {
                    method: "POST",
                    url: contactUrls.deleteContact + "/" + id
                });
            };

            this.getFilteredContacts = function (filter) {
                return $http(
                {
                    method: "POST",
                    url: contactUrls.getFilteredContacts,
                    headers: { 'Content-Type': "application/json" },
                    data: filter
                });
            };

            this.getNumberOfFilteredContacts = function (filter) {
                return $http(
                {
                    method: "POST",
                    url: contactUrls.getNumberOfFilteredContacts,
                    headers: { 'Content-Type': "application/json" },
                    data: filter
                });
            };
        }
    ]
);

File: ContactListWebApp/Services/ContactService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using ContactListWebApp.DTOs;
using ContactListWebApp.Mappers;
using ContactListWebApp.Repositories;

namespace ContactListWebApp.Services
{
    public class ContactService : IContactService
    {
        private readonly IContactRepository _contactRepository;

        public ContactService(IContactRepository contactRepository)
        {
            _contactRepository = contactRepository;
        }

        public async Task<int> AddContact(ContactDto contact)
        {
            try
            {
                return await _contactRepository.AddContact(ContactMapper.MapFromDto(contact));
            }
            catch (ArgumentException)
            {
                throw;
            }
        }

        public async Task DeleteContact(int id)
        {
            try
            {
                await _contactRepository.DeleteContact(id);
            }
            catch (InvalidOperationException)
            {
                throw;
            }
        }

        public Task<int> GetNumberOfFilteredContacts(FilterDto filter)
        {
            return _contactRepository.GetNumberOfFilteredContacts(filter);
        }

        public async Task<IEnumerable<ContactDto>> GetFilteredContacts(FilterDto filter)
        {
            return (await _contactRepository.GetFilteredContacts(filter)).Select(ContactMapper.MapToDto);
        }

        public async Task UpdateContact(ContactDto contact)
        {
            try
            {
                await _contactRepository.UpdateContact(ContactMapper.MapFromDto(contact));
            }
            catch (ArgumentException)
            {
                throw;
            }
            catch (InvalidOperationException)
            {
                throw;
            }
        }
    }
}